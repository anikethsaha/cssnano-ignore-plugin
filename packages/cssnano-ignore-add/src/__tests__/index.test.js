const fs = require('fs');
const path = require('path');
const helperPlugin = require('./helper/_testPlugin.js');
const postcss = require('postcss');
const plugin = require('..');

function run(fixturename, data = []) {
  it(
    'should match the snapshot for ' + fixturename.replace('.css', ''),
    (done) => {
      const input = fs.readFileSync(
        path.join(__dirname, `./fixtures/${fixturename}`),
        'utf8'
      );

      let result = postcss([
        helperPlugin({ data }),
        plugin.default(),
      ]).process(input, { from: undefined });

      expect(result.css).toMatchSnapshot();

      done();
    }
  );
}

fs.readdirSync(path.join(__dirname, './fixtures'))
  .filter((filename) => /(.).css/g.test(filename))
  .forEach((fixtureName) => {
    const data = require(path.join(
      __dirname,
      `./fixtures/${fixtureName.replace('.css', '')}.json`
    ));
    run(fixtureName, data);
  });
