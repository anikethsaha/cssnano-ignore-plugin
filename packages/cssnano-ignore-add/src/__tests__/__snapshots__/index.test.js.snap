// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot for atRule 1`] = `
"@media screen and (min-width: 480px) {
  body {
    background-color: lightgreen;
    /* expects > width : 100% */
    width: 100%;
  }
}
"
`;

exports[`should match the snapshot for basic 1`] = `
"ul li {
  margin: auto;
  /* expects > background-color : red */
  background-color: red;
}
"
`;
